### PR TITLE
Final API contract.

### PR DESCRIPTION
**Pull request by Georgio-Khoury**
The file is added in docs folder.

### CODE REVIEW

File: docs/api-spec-swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,3370 @@
...
+          content:
+           application/json:
+              schema:
+                  type: object
+                  properties:
+                    message:
+                      type: string
+                      example: Resource not found.
+                      
+  /resources/{resourceId}/save:
```
Comment (Hkataya) [inline docs/api-spec-swagger.yaml:1085]: I would move the userid to the url part. /resources/{resourceId}/save/{userId} is more concise.

File: docs/api-spec-swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,3370 @@
...
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: Forbidden - User does not have permission
+                  
+  
+  /jobs/{jobId}{publish}/approve:
```
Comment (Hkataya) [inline docs/api-spec-swagger.yaml:2198]: can we remove publish here?

File: docs/api-spec-swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,3370 @@
...
+        title:
+          type: string
+          example: Technology
+    JobLevel:
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+          example: f21e77c9-8fd6-4d58-ae91-2560b53c2389
```
Comment (Hkataya) [inline docs/api-spec-swagger.yaml:2983]: minor comment the id here is int. Same for Category and type and all non-major entities. major entities are (job, event, resource, company, user).

File: docs/api-spec-swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,3370 @@
...
+  #         content:
+  #           application/json:
+  #             schema:
+  #               type: object
+  #               properties:
+  #                 message:
+  #                   type: string
+  #                   example: Event not found or User not found
+
+  /events/{eventId}{userId}/register:
```
Comment (Hkataya) [inline docs/api-spec-swagger.yaml:1400]: /events/{eventId}/register/{userId}

File: docs/api-spec-swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,3370 @@
...
+                    userId:
+                      type: string
+                      format: uuid
+                      example: f21e77c9-8fd6-4d58-ae91-2560b53c2389
+                    username:
+                      type: string
+      security:
+        - jwt: []
+
+  /events/{eventId}/save:
```
Comment (Hkataya) [inline docs/api-spec-swagger.yaml:1486]: /events/{eventId}/save/{userId}

File: docs/api-spec-swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,3370 @@
...
+          description: Job not found
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: Job not found
+  /jobs/{jobId}/save:
```
Comment (Hkataya) [inline docs/api-spec-swagger.yaml:2326]: should be in jobs section. also /jobs/{jobId}/save/{userId} without request body

**Review by Hkataya — CHANGES_REQUESTED** [review]
Some minor points. Otherwise, nice work!

File: docs/api-spec-swagger.yaml (hunk 2 above)
Comment (Georgio-Khoury) [reply docs/api-spec-swagger.yaml:2198]: Should it be moved to the request body on the Job post endpoint? It indicates whether the job post should be sent to Whatsapp or not.

**Review by Hkataya — APPROVED** [review]

**Pull request merged**